src/sentry/static/sentry/app/views/projectDetail/projectDetail.tsx

```diff
@@ -0,0 +1,141 @@
+import React from 'react';
+import {RouteComponentProps} from 'react-router/lib/Router';
+import styled from '@emotion/styled';
+
+import {t} from 'app/locale';
+import {Organization, Project} from 'app/types';
+import AsyncView from 'app/views/asyncView';
+import GlobalSelectionHeader from 'app/components/organizations/globalSelectionHeader';
+import LightWeightNoProjectMessage from 'app/components/lightWeightNoProjectMessage';
+import {PageContent} from 'app/styles/organization';
+import routeTitleGen from 'app/utils/routeTitle';
+import {IconSettings, IconSiren} from 'app/icons';
+import * as Layout from 'app/components/layouts/thirds';
+import Breadcrumbs from 'app/components/breadcrumbs';
+import Button from 'app/components/button';
+import Access from 'app/components/acl/access';
+import ButtonBar from 'app/components/buttonBar';
+import IdBadge from 'app/components/idBadge';
+import TextOverflow from 'app/components/textOverflow';
+
+type RouteParams = {
+  orgId: string;
+  projectId: string;
+};
+
+type Props = RouteComponentProps<RouteParams, {}> & {
+  organization: Organization;
+};
+
+type State = {
+  project?: Project;
```

Comment:
```suggestion
  project: Project | null
```

we should discuss it again in the next tsc

Reply:
It can also be undefined while initially loading no?

Reply:
the function `getDefaultState` of the `AsynComponent` sets all endpoints key to `null`  😉

Reply:
I don't think so. The function `getDefaultState` sets all the endpoints keys to `null` in the initial loading.